view-and-share/script.js

```diff
@@ -0,0 +1,208 @@
+var connection = null;
+
+var authSettings = {
+  spanButtonID: 'pryv-button', // span id the DOM that will be replaced by the Service specific button
+  onStateChange: pryvAuthStateChange, // event Listener for Authentication steps
+  authRequest: { // See: https://api.pryv.com/reference/#auth-request
+    requestingAppId: 'pryv-example-view-and-share', // to customize for your own app
+    requestedPermissions: [
+      {
+        streamId: 'body',
+        defaultName: 'Body',
+        level: 'read' // permissions for the app to read data in stream 'Body'
+      },
+      {
+        streamId: 'baby',
+        defaultName: 'Baby',
+        level: 'read' // permissions for the app to read data in stream 'Baby'
+      }
+    ],
+    clientData: {
+      'app-web-auth:description': {
+        'type': 'note/txt',
+        'content': 'This sample app demonstrates how you can visualize and share data with an app token.' // to customize according to your own use case
+      }
+    },
+  }
+};
+
+function pryvAuthStateChange(state) { // called each time the authentication state changes
+  console.log('##pryvAuthStateChange', state);
+  if (state.id === Pryv.Browser.AuthStates.AUTHORIZED) {
+    resetData();
+    document.getElementById('please-login').style.visibility = 'hidden';
+    document.getElementById('data-view').style.visibility = 'visible';
+    document.getElementById('sharing-view').style.visibility = 'visible';
+    username = state.displayName; 
+    connection = new Pryv.Connection(state.apiEndpoint);
+    loadData();
+  }
+  if (state.id === Pryv.Browser.AuthStates.INITIALIZED) {
+    document.getElementById('please-login').style.visibility = 'visible';
+    document.getElementById('data-view').style.visibility = 'hidden';
+    document.getElementById('sharing-view').style.visibility = 'hidden';
+    connection = null;
+    resetData();
+  }
+}
+
+// following the APP GUIDELINES: https://api.pryv.com/guides/app-guidelines/
+const urlParams = new URLSearchParams(window.location.search);
+const apiEndpoint = urlParams.get('pryvApiEndpoint');
+const serviceInfoUrl = urlParams.get('pryvServiceInfoURL') || 'https://reg.pryv.me/service/info';
+
+var service = null; // will be initialized after setupAuth;
+var username = null; // will be inialized after AUTHORIZED auth State is received
+window.onload = async (event) => {
+  
+  if (apiEndpoint) { // if apiEndpoint then we are in "View only mode"
+    document.getElementById('welcome-message-mme').style.visibility = 'hidden';
+    document.getElementById('please-login').style.visibility = 'hidden';
+    document.getElementById('data-view').style.visibility = 'visible';
+    document.getElementById('welcome-message-viewer').style.visibility = 'visible';
+    document.getElementById('username').innerText = apiEndpoint.split('@')[1].slice(0,-1);
```

Comment:
there might be an issue, it cuts the last param of the API endpoint when provided without the trailer`/`: https://api.pryv.com/app-web-examples/view-and-share/?pryvApiEndpoint=https://ckbry9n6h009o1od3qiv3qz7u@mariana.pryv.me